staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -114,6 +115,15 @@ func (a customResourceStrategy) PrepareForCreate(ctx context.Context, obj runtim
 	}
 
 	accessor, _ := meta.Accessor(obj)
+	if len(accessor.GetAnnotations()[genericapirequest.AnnotationKey]) > 0 {
+		// in general the shard annotation is not attached to objects, instead, it is assigned by the storage layer on the fly
+		// to avoid an additional UPDATE request (mismatch on the generation field) replicated objects have the shard annotation set
+		// thus we need to remove the shard annotation and simply return early so that the generation is not reset to 1
+		annotations := accessor.GetAnnotations()
+		delete(annotations, genericapirequest.AnnotationKey)
+		accessor.SetAnnotations(annotations)
```

Comment:
I don't believe you need to get/set the map. You should be able to delete directly. https://go.dev/play/p/r7Wr8PbmwN9. Hopefully meta.Accessor doesn't change that?

Reply:
ah, right, recently I have worked with an unstructured object that gave me a copy, here indeed we get a pointer.